algorithms/scripts/RIT1-PCA-htcai.py

```diff
@@ -65,54 +63,78 @@
 
 # In[8]:
 
-# Here are the percentage of tumors with RIT1
-y.value_counts(True)
+# Here is the count of tumors with RIT1
+y.value_counts()
 
 
-# ## Set aside 20% of the data for testing
+# ## Set aside 30% of the data for testing
 
 # In[9]:
 
 # Typically, this can only be done where the number of mutations is large enough
-X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, stratify=y, random_state=0)
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, stratify=y, random_state=0)
 'Size: {:,} features, {:,} training samples, {:,} testing samples'.format(len(X.columns), len(X_train), len(X_test))
 
 
-# ## Feature Standardization
-
 # In[10]:
 
-get_ipython().run_cell_magic('time', '', 'scale_pre = StandardScaler()\nX_train_scale = scale_pre.fit_transform(X_train)\nX_test_scale = scale_pre.transform(X_test)')
+# Here is the count of tumors with RIT1 in the testing data
+y_test.value_counts()
 
 
-# ## Reducing noise via PCA
+# ## Preprocessing and noise reduction
 
 # In[11]:
 
-get_ipython().run_cell_magic('time', '', 'n_components = 30\npca = PCA(n_components=n_components, random_state=0)\nX_train_pca = pca.fit_transform(X_train_scale)\nX_test_pca = pca.transform(X_test_scale)')
+scale_pre = StandardScaler()
+X_train_scale = scale_pre.fit_transform(X_train)
+X_test_scale = scale_pre.transform(X_test)
 
 
 # In[12]:
 
-get_ipython().run_cell_magic('time', '', 'scale_post = StandardScaler()\nX_train_scale = scale_post.fit_transform(X_train_pca)\nX_test_scale = scale_post.transform(X_test_pca)')
-
+n_components = 35
+pca = PCA(n_components=n_components, random_state=0)
+X_train_pca = pca.fit_transform(X_train_scale)
+X_test_pca = pca.transform(X_test_scale)
 
-# ## Parameters
 
 # In[13]:
 
+# Percentage of preserved variance
+print('{:.4}'.format(sum(pca.explained_variance_ratio_)))
+
+
+# In[14]:
+
+scale_post = StandardScaler()
+X_train_scale = scale_post.fit_transform(X_train_pca)
+X_test_scale = scale_post.transform(X_test_pca)
+
+
+# ## Parameters and Classifier Fitting
+
+# In[15]:
+
 param_grid = {
-    'alpha': [10 ** x for x in range(-4, 1)],
-    'l1_ratio': [0, 0.2, 0.5, 0.8, 1],
+    'alpha': [2**x for x in range(-20, 60)],
+    'l1_ratio': [0]
 }
 
 
-# In[14]:
+# In[16]:
+
+sss = StratifiedShuffleSplit(n_splits=10, test_size=0.3, random_state=0)
```

Comment:
If computation time doesn't become prohibitive, I'd love something more thorough such as:

```python
StratifiedShuffleSplit(n_splits=100, test_size=0.1, random_state=0)
```

`n_splits` times `test_size` equals the average number of times each observation will be used for testing in CV. The higher this is the less variation we should have in our performance estimates. However, we want `test_size` to be low, so the percent of observations used for training is maximized. Therefore, we should increase `n_splits` up until the point where it starts taking a long time. Make sense?